sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/resource_configuration.py

```diff
@@ -6,7 +6,7 @@
 import logging
 from typing import Any, Dict, Optional
 
-from azure.ai.ml._restclient.v2022_10_01.models import ResourceConfiguration as RestResourceConfiguration
+from azure.ai.ml._restclient.v2023_04_01_preview.models import ResourceConfiguration as RestResourceConfiguration
```

Comment:
@luigiw QQ: 'max_instance_count' appears in `v2023_04_01_preview.models.ResourceConfiguration` why don't we have it in the non-preview package? (v2023_04_01.models.ResourceConfiguration). Is it ok to import from this here?

Reply:
JobOperations switched to April preview. https://github.com/Azure/azure-sdk-for-python/blob/3f3c1eee5241f54836cccad2340e5af7352a4b72/sdk/ml/azure-ai-ml/azure/ai/ml/_ml_client.py#L336 

Let's switch all job imports to April preview.

Reply:
You mean replace `v2022_10_01` with `v2023_04_01_preview` across the board? There's about 71 files in all.

Reply:
@luigiw please review again

Reply:
(most of the job related imports were updated, but i guess there were some that were missed)